lib/src/migrators/module.dart

```diff
@@ -43,14 +43,17 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   ///
   /// We store all declarations instead of just the most recent one for use in
   /// detecting configurable variables.
-  final _globalVariables = normalizedMap<List<VariableDeclaration>>();
+  final _globalVariables = normalizedMap<VariableDeclaration>();
 
   /// Global mixins defined at any time during the migrator run.
   final _globalMixins = normalizedMap<MixinRule>();
 
   /// Global functions defined at any time during the migrator run.
   final _globalFunctions = normalizedMap<FunctionRule>();
 
+  /// Stores whether a given VariableDeclaration has been referenced.
+  final _variableReferenced = <VariableDeclaration, bool>{};
```

Comment:
```suggestion
  final _referencedVariables = Set<VariableDeclaration>();
```

Maps with boolean keys are just Sets with weirder APIs, imo.